### PR TITLE
Feature/tag proposals

### PR DESCRIPTION
**Pull request by eheinrich**
Allow users submitting a science application to select whether an application is for a student thesis. Besides that:

- If a proposal has tags, those tags are displayed on the proposal detail page
- Another filter has been added to the proposal list page to allow a user to filter for a specific proposal tag

This branch is running here http://172.16.4.73:8080/ (corresponding backend running here http://172.16.4.73:8000/). The backend has a snapshot of prod so you should be able to log in with you usual credentials. I've already created a science application and proposal with a tag so you can see how those look.

### CODE REVIEW

File: src/components/EditSciApplication.vue [hunk 1]
```diff
@@ -37,6 +37,14 @@
             field-type="textarea"
             rows="8"
           ></basic-custom-field>
+          <basic-custom-field
+            v-model="isStudent"
+            field-type="checkbox"
+            field="is-student-tag"
+            :label="''"
+            :errors="apiValidationErrors.tags"
+            checkbox-option-label="The data will be an important component of a PhD thesis"
```
Comment (jnation3406) [inline src/components/EditSciApplication.vue:46]: "Data from this proposal will be an..."

File: src/components/EditSciApplication.vue [hunk 2]
```diff
@@ -579,8 +588,13 @@ export default {
         pi_institution: initialSciApp.pi_institution || '',
         coinvestigator_set: initialSciApp.coinvestigator_set || [],
         timerequest_set: initialSciApp.timerequest_set || [],
+        tags: initialSciApp.tags || [],
         pdf: null
       };
+      let isStudent = false;
+      if (sciApp.tags.indexOf('student') >= 0) {
```
Comment (jnation3406) [inline src/components/EditSciApplication.vue:595]: why not just `sciApp.tags.includes('student')`?

File: src/views/ProposalList.vue [hunk 3]
```diff
@@ -31,11 +31,13 @@
...
         <b-form inline @submit="onSubmit">
-          <label class="mr-sm-2" for="input-proposal-active">Proposal is active:</label>
+          <label class="mr-1" for="input-proposal-active">Proposal is active:</label>
           <b-form-select id="input-proposal-active" v-model="queryParams.active" :options="proposalActiveOptions" />
-          <label class="m-sm-2" for="input-proposal-semester">Semester:</label>
+          <label class="m-1" for="input-proposal-semester">Semester:</label>
           <b-form-select id="input-proposal-semester" v-model="queryParams.semester" :options="semesterOptions"></b-form-select>
-          <b-button type="submit" class="m-sm-1" variant="outline-primary" :disabled="isBusy">Filter</b-button>
+          <label for="select-proposal-tag" class="m-1">Proposal tag:</label>
+          <b-form-select id="select-proposal-tag" v-model="queryParams.tag" :options="tagOptions" />
```
Comment (jnation3406) [inline src/views/ProposalList.vue:39]: this should probably be a multi-select if we change the filter to be a multiple choice filter

**Review by jnation3406 — APPROVED** [review]
looks good

File: src/components/EditSciApplication.vue (hunk 2 above)
Comment (eheinrich) [reply src/components/EditSciApplication.vue]: Done

File: src/components/EditSciApplication.vue (hunk 1 above)
Comment (eheinrich) [reply src/components/EditSciApplication.vue]: Done

**Pull request merged**